src/nextjournal/clerk/webserver.clj

```diff
@@ -223,13 +240,25 @@
 (defn show! [opts file-or-ns]
   ((resolve 'nextjournal.clerk/show!) opts file-or-ns))
 
+(defn route-index
+  "A routing function"
+  [{:as opts :keys [index expanded-paths]} nav-path]
+  (if (str/blank? nav-path)
+    (or index
+        (get (set expanded-paths) (maybe-add-extension "index"))
+        "'nextjournal.clerk.index")
+    nav-path))
+
 (defn navigate! [{:as opts :keys [nav-path]}]
+  ;; TODO: perform `route-index` when needed, needs `:expanded-paths` for `v/route-index?`
   (show! opts (->file-or-ns (maybe-add-extension nav-path))))
 
 (defn prefetch-request? [req] (= "prefetch" (-> req :headers (get "purpose"))))
 
 (defn serve-notebook [{:as req :keys [uri]}]
-  (let [nav-path (subs uri 1)]
+  (let [opts (process-paths @!server)
```

Comment:
Maybe call `process-paths` once within `serve!` below instead of each time here? `process-path` reads the filesystem if I can see correctly.

Reply:
Yes, calling this intentionally every time so changes to the filesystem are picked up without requiring a restart.